src/main/java/com/sigma/cinematicketpro/util/JwtTokenUtils.java

```diff
@@ -0,0 +1,74 @@
+package com.sigma.cinematicketpro.util;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.security.Key;
+import java.time.Duration;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Component
+public class JwtTokenUtils {
+    @Value("${spring.jwt.secret}")
+    private String secret;
+    @Value("${spring.jwt.lifetime}")
+    private Duration jwtLifetime;
+
+    public String generateToken(UserDetails userDetails) {
+        Map<String, Object> claims = getDefaultClaims(userDetails);
+        return generateToken(userDetails, claims);
+    }
+
+    public String generateToken(UserDetails userDetails, Map<String, Object> extraClaims) {
+        return Jwts
+                .builder()
+                .setClaims(extraClaims)
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + jwtLifetime.toMillis()))
+                .signWith(getSignInKey(), SignatureAlgorithm.HS256)
+                .setSubject(userDetails.getUsername())
+                .compact();
+    }
+
+    public String getUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
+        Claims allClaims = extractAllClaims(token);
+        return claimsResolver.apply(allClaims);
+    }
+
+    public boolean isTokenValid(String token, UserDetails userDetails) {
+        return getUsername(token).equals(userDetails.getUsername());
+    }
+
+    public Claims extractAllClaims(String token) {
+        return Jwts
+                .parserBuilder()
+                .setSigningKey(getSignInKey())
+                .build()
+                .parseClaimsJws(token)
+                .getBody();
+    }
+
+    private Key getSignInKey() {
+        byte[] keyBytes = Decoders.BASE64.decode(secret);
+        return Keys.hmacShaKeyFor(keyBytes);
+    }
+
+    private Map<String, Object> getDefaultClaims(UserDetails userDetails) {
+        Map<String, Object> claims = new HashMap<>();
+        claims.put("roles", userDetails.getAuthorities());
+        return claims;
+    }
+}
```

Comment:
looks like you cope pasted all this class from some where - refactor it normally

Reply:
fixed